File: problems_026to050/problem_039/solution.py
import os
import sys
from math import ceil

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = None
        max_triangle_count = 0
        for p in range(12, 1001, 2):
            triangle_count = 0
            max_a = ceil(p / 3) - 1
            for a in range(max_a, 0, -1):
                for b in range(a+1, ceil((p-a) / 2)):
                    c = p - (a + b)
                    if a**2 + b**2 == c**2:
                        triangle_count += 1
            if max_triangle_count < triangle_count:
                max_triangle_count = triangle_count
                answer = p
        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_026to050/problem_039/solution.js
const solution = () => {
  let answer = null;

  let maxTriangleCount = 0;

  for (let p = 12; p <= 1000; p += 2) {
    let triangleCount = 0;
    const maxA = Math.ceil(p / 3) - 1;

    for (let a = maxA; a > 0; a -= 1) {
      for (let b = a + 1; b < Math.ceil((p - a) / 2); b += 1) {
        const c = p - (a + b);

        if (a ** 2 + b ** 2 === c ** 2) {
          triangleCount += 1;
        }
      }
    }

    if (maxTriangleCount < triangleCount) {
      maxTriangleCount = triangleCount;
      answer = p;
    }
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
